gno.land/cmd/gnoland/config_set.go

```diff
@@ -0,0 +1,253 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gnolang/gno/tm2/pkg/bft/config"
+	"github.com/gnolang/gno/tm2/pkg/bft/state/eventstore/types"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+)
+
+var errInvalidConfigSetArgs = errors.New("invalid number of config set arguments provided")
+
+// newConfigSetCmd creates the config set command
+func newConfigSetCmd(io commands.IO) *commands.Command {
+	cfg := &configCfg{}
+
+	cmd := commands.NewCommand(
+		commands.Metadata{
+			Name:       "set",
+			ShortUsage: "config set <key> <value>",
+			ShortHelp:  "Edits the Gno node configuration",
+			LongHelp: "Edits the Gno node configuration at the given path " +
+				"by setting the option specified at <key> to the given <value>",
+		},
+		cfg,
+		func(_ context.Context, args []string) error {
+			return execConfigEdit(cfg, io, args)
+		},
+	)
+
+	return cmd
+}
+
+func execConfigEdit(cfg *configCfg, io commands.IO, args []string) error {
+	// Load the config
+	loadedCfg, err := config.LoadConfigFile(cfg.configPath)
+	if err != nil {
+		return fmt.Errorf("unable to load config, %w", err)
+	}
+
+	// Make sure the edit arguments are valid
+	if len(args) != 2 {
+		return errInvalidConfigSetArgs
+	}
+
+	var (
+		key   = args[0]
+		value = args[1]
+	)
+
+	// Update the config field
+	if err := updateConfigField(
+		loadedCfg,
+		key,
+		value,
+	); err != nil {
+		return fmt.Errorf("unable to update config field, %w", err)
+	}
+
+	// Make sure the config is now valid
+	if err := loadedCfg.ValidateBasic(); err != nil {
+		return fmt.Errorf("unable to validate config, %w", err)
+	}
+
+	// Save the config
+	if err := config.WriteConfigFile(cfg.configPath, loadedCfg); err != nil {
+		return fmt.Errorf("unable to save updated config, %w", err)
+	}
+
+	io.Printfln("Updated configuration saved at %s", cfg.configPath)
+
+	return nil
+}
+
+// updateFieldAtPath updates the field at the given path, with the given value
+func updateConfigField(config *config.Config, key, value string) error {
+	// Get the config value using reflect
+	configValue := reflect.ValueOf(config).Elem()
+
+	// Get the value path, with sections separated out by a period
+	path := strings.Split(key, ".")
+
+	field, err := getFieldAtPath(configValue, path)
+	if err != nil {
+		return err
+	}
+
+	// We've reached the actual field, check if it can be updated
+	if !field.CanSet() {
+		return fmt.Errorf("unable to set value")
+	}
+
+	// Convert the value to the field's type
+	v, err := convertStringToType(value, field.Interface())
+	if err != nil {
+		return fmt.Errorf("unable to convert value to field type, %w", err)
+	}
+
+	// Update the field value
+	field.Set(reflect.ValueOf(v))
+
+	return nil
+}
+
+// generateInvalidFieldError generates an invalid field error
+func generateInvalidFieldError(field string, value reflect.Value) error {
```

Comment:
```suggestion
// newInvalidFieldError creates an error for non-existent struct fields
// being passed as arguments to [getFieldAtPath].
func newInvalidFieldError(field string, value reflect.Value) error {
```

BTW suggest moving this to config.go together with getFieldAtPath

Reply:
Agreed:

[c8aca4a](https://github.com/gnolang/gno/pull/1605/commits/c8aca4adf75eb9e2a6c4629a921b996c91682eab)